src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedAnalyzer.cs

```diff
@@ -0,0 +1,566 @@
+        private static readonly LocalizableString s_localizableApplicationDescription = CreateLocalizableResourceString(nameof(ConstantExpectedApplicationDescription));
+        private static readonly LocalizableString s_localizableUsageTitle = CreateLocalizableResourceString(nameof(ConstantExpectedUsageTitle));
+        private static readonly LocalizableString s_localizableUsageDescription = CreateLocalizableResourceString(nameof(ConstantExpectedUsageDescription));
+
+        internal static class CA1860
+        {
+            internal const string Id = nameof(CA1860);
+            internal const RuleLevel Level = RuleLevel.BuildError;
+            internal static readonly DiagnosticDescriptor UnsupportedTypeRule = DiagnosticDescriptorHelper.Create(
+                Id,
+                s_localizableApplicationTitle,
+                CreateLocalizableResourceString(nameof(ConstantExpectedNotSupportedMessage)),
+                DiagnosticCategory.Performance,
+                Level,
+                description: s_localizableApplicationDescription,
+                isPortedFxCopRule: false,
+                isDataflowRule: false);
+
+            internal static readonly DiagnosticDescriptor IncompatibleConstantTypeRule = DiagnosticDescriptorHelper.Create(
+                Id,
+                s_localizableApplicationTitle,
+                CreateLocalizableResourceString(nameof(ConstantExpectedIncompatibleConstantTypeMessage)),
+                DiagnosticCategory.Performance,
+                Level,
+                description: s_localizableApplicationDescription,
+                isPortedFxCopRule: false,
+                isDataflowRule: false);
+
+            internal static readonly DiagnosticDescriptor InvalidBoundsRule = DiagnosticDescriptorHelper.Create(
+                Id,
+                s_localizableApplicationTitle,
+                CreateLocalizableResourceString(nameof(ConstantExpectedInvalidBoundsMessage)),
+                DiagnosticCategory.Performance,
+                Level,
+                description: s_localizableApplicationDescription,
+                isPortedFxCopRule: false,
+                isDataflowRule: false);
+
+            internal static readonly DiagnosticDescriptor InvertedRangeRule = DiagnosticDescriptorHelper.Create(
+                Id,
+                s_localizableApplicationTitle,
+                CreateLocalizableResourceString(nameof(ConstantExpectedInvertedRangeMessage)),
+                DiagnosticCategory.Performance,
+                Level,
+                description: s_localizableApplicationDescription,
+                isPortedFxCopRule: false,
+                isDataflowRule: false);
+        }
+
+        internal static class CA1861
+        {
+            internal const string Id = nameof(CA1861);
+            internal const RuleLevel Level = RuleLevel.BuildWarning;
```

Comment:
There is a requirement to add [performance test](https://github.com/dotnet/roslyn-analyzers/tree/main/src/PerformanceTests) for `BuildWarning` and above level analyzers, I would not require that with this PR, only if you can add (its not difficult, see the [examples](https://github.com/dotnet/roslyn-analyzers/tree/main/src/PerformanceTests/Tests/Enabled))

Reply:
Add the tests, i created a separate test for each id. Not sure that's correct. Also I'm not sure how many diagnostics should be triggerred